mlflow-tests/src/mlflow_tests/manager/user.py

```
@@ -0,0 +1,144 @@
+                    verb=test_verb,
+                    max_retries=10,
+                    retry_delay=1.0
+                )
+                logger.info(f"RBAC verification passed for {name} - can {test_verb} {resource.get_k8s_resource()}")
+            except Exception as e:
+                logger.error(f"RBAC verification failed for user '{name}': {e}")
+                raise
+
+        logger.info(f"User '{name}' now has {verb_strings} access to {len(resources)} resource types in namespace '{workspace_name}'")
+
+    def delete_user(self, username: str, namespace: str = None) -> None:
```

Ruff (0.15.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)